TOOLS/dylib-unhell.py

```diff
@@ -157,6 +162,104 @@ def process_swift_libraries(binary):
+        os.path.join(os.path.expanduser("~"), ".config/vulkan/icd.d"),
+        "/etc/xdg/vulkan/icd.d",
+        "/usr/local/etc/vulkan/icd.d",
+        "/etc/vulkan/icd.d",
+        os.path.join(os.path.expanduser("~"), ".local/share/vulkan/icd.d"),
+        "/usr/local/share/vulkan/icd.d",
+        "/usr/share/vulkan/icd.d",
+    ]
+    vulkanIcdPath = os.path.join(resources_path(binary), "vulkan/icd.d")
+    vulkanIcdJsonPath = os.path.join(vulkanIcdPath, "MoltenVK_icd.json")
+    systemDriverPath = ""
+
+    for driverPath in vulkanDriverSearchPaths:
+        if os.path.exists(driverPath):
+            systemDriverPath = driverPath
+            break
+
+    if not systemDriverPath:
+        print(">>> could not find vulkan driver")
+        return
+
+    print(">>> copying vulkan icd.d folder to bundle")
+    copy_tree(systemDriverPath, vulkanIcdPath)
+
+    if os.path.exists(vulkanIcdJsonPath):
+        print(">>> modifiying vulkan icd layer json MoltenVK_icd.json")
+        vulkanIcdJsonFile = open(vulkanIcdJsonPath, 'r')
+        vulkanIcdJsonData = json.load(vulkanIcdJsonFile)
+        oldIcdLibPath = vulkanIcdJsonData["ICD"]["library_path"]
+        vulkanIcdJsonData["ICD"]["library_path"] = "../../../Frameworks/libMoltenVK.dylib"
+        vulkanIcdJsonFile = open(vulkanIcdJsonPath, 'w')
+        json.dump(vulkanIcdJsonData, vulkanIcdJsonFile, indent=4)
+
+        print(">>> copying vulkan driver library libMoltenVK.dylib")
+        systemDriverLibPath = os.path.join(systemDriverPath, oldIcdLibPath)
+        if os.path.exists(systemDriverLibPath):
+            frameWorkPath = os.path.join(vulkanIcdPath, "../../../Frameworks")
+            if not os.path.exists(frameWorkPath):
+                os.makedirs(frameWorkPath)
+            shutil.copy(systemDriverLibPath, os.path.join(frameWorkPath, "libMoltenVK.dylib"))
+
+def process_vulkan_layer_libraries(binary):
+    # layer serach paths https://github.com/KhronosGroup/Vulkan-Loader/blob/main/docs/LoaderLayerInterface.md#macos-layer-discovery
+    vulkanLayerSearchPaths = [
+        os.path.join(os.path.expanduser("~"), ".config/vulkan/explicit_layer.d"),
+        "/etc/xdg/vulkan/explicit_layer.d",
+        "/usr/local/etc/vulkan/explicit_layer.d",
+        "/etc/vulkan/explicit_layer.d",
+        os.path.join(os.path.expanduser("~"), ".local/share/vulkan/explicit_layer.d"),
+        "/usr/local/share/vulkan/explicit_layer.d",
+        "/usr/share/vulkan/explicit_layer.d",
+    ]
+
+    vulkanLayerPath = os.path.join(resources_path(binary), "vulkan/explicit_layer.d")
+    systemLayerPath = ""
+
+    for layerPath in vulkanLayerSearchPaths:
+        if os.path.exists(layerPath):
+            systemLayerPath = layerPath
+            break
+
+    if not systemLayerPath:
+        print(">>> could not find vulkan layers")
+        return
+
+    if not os.path.exists(vulkanLayerPath):
+        os.makedirs(vulkanLayerPath)
+
+    vulkanLayers = [
+        "VkLayer_khronos_synchronization2.json"
```

Comment:
Is this really needed? synchronization2 was added to MoltenVK in 1.2.6 and libplacebo doesn't require it anyway.

Reply:
```
13:40	der_richter	https://0x0.st/Hn2E.36.png
					do i need any of those explicit vulkan layers for mpv or are they useful?
					asking since i am wondering what i should bundle with a macOS app bundle
15:22	haasn		der_richter: we require synchronization2
					I think mvk only implements that via layers
```

according to haasn we need it.

Reply:
Well, that’s not the case for a long time. See https://github.com/KhronosGroup/MoltenVK/pull/2021

Reply:
just to clarify, there are two separate things here.
1. "libplacebo doesn't require it anyway" which is wrong, since haasn explicitly mentioned we need it.
2. "Is this really needed? synchronization2 was added to MoltenVK in 1.2.6" so we don't need it as a vulkan layer anymore

some points for 2:
- anyone using an older version of moltenvk will need the layer(?) (i am still on such an old version)
- newer version of the vulkan sdk support synchronization2 directly through moltenvk but also still provide the layer, which would lead to an unnecessary inclusion
- if we still want to support older moltenvk versions we need to keep it
- for newer version wer don't want to unnecessarily include that layer
- it might be cool do have the layer functionality in the bundle script, if we ever need another layer

does the inclusion of the layer have any negative effect other than being there unnecessarily, or is it just ignored?
do we need a version check on the bundle script?

Reply:
> "libplacebo doesn't require it anyway" which is wrong, since haasn explicitly mentioned we need it.

To clear this up. synchronization2 is NOT required in libplacebo after this commit https://github.com/haasn/libplacebo/commit/1d7092334dead779c42f15fd66b01e3086d472f6 I think @haasn forgot about it. (PR with related commits https://code.videolan.org/videolan/libplacebo/-/merge_requests/499)

So as of now, we don't need to include synchronization2 layer as it is workaround directly in libplacebo. Alternative is to make it required again and then force certain MVK version or use of layer in other case. @haasn thoughts?

Reply:
so it is an [optional recommended feature](https://github.com/haasn/libplacebo/commit/1d7092334dead779c42f15fd66b01e3086d472f6#diff-2522421490bb26636ab0d7699adc9837f077e59ef569f524e196e1301de1c3ffR228-R235). so we should copy the loader over for at least older versions.

i made some adjustments. so there shouldn't be a problem anymore.